### PR TITLE
Deploying RESTful APIs using Node, Express 4 to Kubernetes clusters

### PR DESCRIPTION
**Pull request by jaredphelix**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by jaredphelix** [comment]
opening my first article..

File: content/articles/restful-node-deployment-using-docker/index.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+- [Dockerizing the Express application](#dockerizing-the-express-application)
+- [Setup YAML service to deploy Dockerized Node Express application](#setup-yaml-service-to-deploy-dockerized-node-express-application)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Prerequisites
+
+To follow along with this tutorial, you need the following:
+
+- Node.js downloaded and installed in your local development environment.
```
Comment (paulodhiambo) [inline content/articles/restful-node-deployment-using-docker/index.md:22]: Add Node.js download link

File: content/articles/restful-node-deployment-using-docker/index.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+RUN mkdir -p /usr/src/app  
+WORKDIR /usr/src/app  
+COPY . /usr/src/app
+
+EXPOSE 8000  
+RUN npm install  
+CMD ["npm", "start"]  
+
+```
```
Comment (paulodhiambo) [inline content/articles/restful-node-deployment-using-docker/index.md:194]: Explain the commands on the Dockerfile

File: content/articles/restful-node-deployment-using-docker/index.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+  links:
+  - mongodb
+  
+  environment:
+    - NODE_ENV=production
+    - MONGODB_ADDRESS=mongodb
+mongodb:  
+  image: mongo
+
+```
```
Comment (paulodhiambo) [inline content/articles/restful-node-deployment-using-docker/index.md:215]: Give description of the entries on this file.

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@jaredphelix amazing tutorial. Review the requested changes and ping me when done.

**Comment by paulodhiambo** [comment]
@hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
@jaredphelix why was this PR closed on May 17th??

**Comment by hectorkambow** [comment]
@jaredphelix FYI PRs can only be created (and submitted) once the articles are completed.

Please refrain from creating a PR to close and then re-open. Thank you.

Cc @paulodhiambo

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@paulodhiambo  I'll mention this to you as well - please do not review PRs that were closed and re-opened. 
Thank you 👍 

Any PR like that (will be asked to be re-submitted) PRs must be only submitted when ready. 💯

**Comment by paulodhiambo** [comment]
> @paulodhiambo I'll mention this to you as well - please do not review PRs that were closed and re-opened.
> Thank you 
> 
> Any PR like that (will be asked to be re-submitted) PRs must be only submitted when ready.

**Noted**

**Comment by hectorkambow** [comment]
@jaredphelix What TOPIC form is this PR tied to??

**Comment by jaredphelix** [comment]
@paulodhiambo, I made minor change on the title to reflect on  then minor edit on the content.

**Comment by hectorkambow** [comment]
great work gents @jaredphelix 
cc @paulodhiambo

**Pull request merged**